src/plugins/intel_gpu/src/plugin/ops/eye.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/op/eye.hpp"
+
+#include <memory>
+
+#include "intel_gpu/plugin/common_utils.hpp"
+#include "intel_gpu/plugin/program.hpp"
+#include "intel_gpu/primitives/eye.hpp"
+
+namespace ov {
+namespace runtime {
+namespace intel_gpu {
+
+namespace {
+
+static void CreateEyeOp(Program& p, const std::shared_ptr<ngraph::op::v9::Eye>& op) {
+    p.ValidateInputs(op, {3, 4});
+
+    const InferenceEngine::SizeVector& output_shapes = op->get_output_shape(0);
+    auto os_sz = output_shapes.size();
+    /*static constexpr*/ size_t dim_size = 4;
+    // convert everything to bfyx format
+    InferenceEngine::SizeVector dims(dim_size, 1);
+    for (size_t i = dim_size, j = os_sz; i > 0 && j > 0; --i, --j) {
+        if (i == 1 && os_sz == 5) {
+            assert(j == 2);
+            dims[i - 1] = output_shapes[j - 1] * output_shapes[j - 2];
+        } else {
+            dims[i - 1] = output_shapes[j - 1];
+        }
+    }
+    auto input_primitives = p.GetInputPrimitiveIDs(op);
+    auto output_shape = tensor_from_dims(dims);
+    auto eye_prim = cldnn::eye(layer_type_name_ID(op), input_primitives, output_shape, op->get_friendly_name());
+    const ngraph::op::v0::Constant* constant = dynamic_cast<const ngraph::op::v0::Constant*>(op->get_input_node_ptr(2));
+
+    switch (constant->get_element_type()) {
+    case ov::element::Type_t::i32:
+        eye_prim.shift = *constant->get_data_ptr<int32_t>();
+        break;
+    case ov::element::Type_t::i64:
+        eye_prim.shift = *constant->get_data_ptr<int64_t>();
+        break;
+    default:
+        throw std::runtime_error{"Input type can be only either i32 or i64"};
+        break;
+    }
+    const ov::element::Type ov_type = op->get_out_type();
+    switch (ov_type) {
+    case ov::element::Type_t::f32:
+        eye_prim.output_data_type = cldnn::type_to_data_type<float>::value;
+        break;
+    case ov::element::Type_t::f16:
+        eye_prim.output_data_type = cldnn::type_to_data_type<cldnn::half_t>::value;
+        break;
+    case ov::element::Type_t::i8:
+        eye_prim.output_data_type = cldnn::type_to_data_type<std::int8_t>::value;
+        break;
+    case ov::element::Type_t::i32:
+        eye_prim.output_data_type = cldnn::type_to_data_type<std::int32_t>::value;
+        break;
+    case ov::element::Type_t::i64:
+        eye_prim.output_data_type = cldnn::type_to_data_type<std::int64_t>::value;
```

Comment:
How about set layout which contains all info : format, shape, type, instead of setting them separately? 

BTW, should eylike operation only supports 4D output?

Reply:
Also if the fixed output rank (4d) is not same as original rank, then we'll need reorder after this operation.